internal/app/transport/watch.go

```diff
@@ -0,0 +1,54 @@
+package transport
+
+import (
+	"fmt"
+
+	gcp "github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+)
+
+// Watch interface abstracts v2 and v3 watches
+type Watch interface {
+	Close()
+	GetCh() interface{}
```

Comment:
Hmm I prefer to avoid the type conversion from `interface{}` where we can. Anything we can do to encapsulate this since this is either of type `chan v2.Response` or type `chan v3.Response`?

Reply:
we could take either approach. We could expose 2 methods instead with concrete signature if you prefer. Let me do that.

Reply:
let me know if the two methods look better now.

Reply:
Would something like this work? I think with the two methods the interface becomes less valuable because it starts to get an understanding of the v2/v3 implementation.

```
func (watch *watchV2) GetChan() ResponseCh { return Response{V2: watch.out} }
func (watch *watchV3) GetChan() ResponseCh { return Response{V3: watch.out} }

type ResponseCh struct {
     V2 chan gcpV2.Response
     V3 chan gcpV3.Response
}

type Watch interface {
    Close()
    GetCh() ResponseCh
    ...
}
```

From the caller:
```
w.GetCh().V2
```

Would suggest doing something similar with `Send(interface{})` if it works.

Reply:
It won't be possible with `Send`, because it is going to be called from version agnostic code in orchestrator. You are right about the inefficacy of the two methods, and that was the purpose of using interface earlier. We are going to add the Request/Response abstractions in the next PR and could show how the interfaces are going to interact. The watch would be called only from the transport package only.

Reply:
If it helps, I can put out a bigger PR about how Request/Response abstractions are going to come through in the subsequent PRs and how the usage of interface will be limited to transport package only, and will be unit tested.

Reply:
I've made changes to include all interfaces, Request/Response/Watch to see the interfaces better. We can iterate on these changes now. Hopefully the changes aren't too big.